tests.py

```diff
@@ -24,33 +34,57 @@ def run_vsm(self, name, input_data, expected_output, use_initial=True):
             cmd += ['--initial-state={}'.format(initial_state)]
 
         cmd += [conf]
-
-        process = Popen(cmd, stdin=PIPE, stdout=PIPE)
-
-        output, _ = process.communicate(input=input_data.encode(), timeout=2)
-
-        self.assertEqual(output.decode() , expected_output)
+        if use_ipc:
+            cmd += [ '--ipc-module=zeromq', '--no-loop' ]
+
+        with Popen(cmd, stdin=PIPE, stdout=PIPE) as proc:
+            if use_ipc:
+                t = threading.Thread(target=send, args=input_data)
```

Comment:
What do you need a thread for? You can send all of your input first, then read all of the output, no?